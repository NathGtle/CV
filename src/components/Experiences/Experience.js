import React from 'react'

const Experience = ({ experience }) => (
  <article className="exp">
    <div className="headlineArticle">
      {experience.entreprise} ({experience.lieu})
    </div>
    <div className="duration">
      {experience.dateDebut} Ã  {experience.dateFin}
    </div>
    <div className="clear" />
    <div className="intitule">{experience.intitule}</div>
    <div className="description display-linebreak">
      {experience.description}
    </div>
    <div className="technologies">{experience.technologies}</div>
  </article>
)

export default Experience
