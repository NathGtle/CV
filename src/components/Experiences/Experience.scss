.exp {
  display: grid;
  grid-template-columns: repeat(10, 1fr);
}

.headline {
  grid-column: 1 / span 8;
}

.duration {
  grid-column-end: 12;
}

.intitule,
.description,
.technologies {
  grid-column: span 12;
}

.intitule {
  color: $primary-color;
  font-size: 1.6rem;
  font-weight: bold;
}

.technologies {
  color: $primary-color;
  font-weight: bold;
  font-size: 1.3rem;
}
