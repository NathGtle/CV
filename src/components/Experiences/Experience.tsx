import React from 'react'
import PropTypes from 'prop-types'
import './Experience.scss'

interface ExperienceInterface {
  experience: {
    entreprise: string
    dateDebut: string
    dateFin: string
    intitule: string
    description: string
    technologies: string
  }
}

const Experience = ({ experience }: ExperienceInterface) => {
  if (!experience) return null

  return (
    <article className="exp">
      <div className="headline">{experience.entreprise}</div>
      <div className="duration">
        {experience.dateDebut} Ã  {experience.dateFin}
      </div>
      <div className="intitule">{experience.intitule}</div>
      <div className="description display-linebreak">{experience.description}</div>
      <div className="technologies">{experience.technologies}</div>
    </article>
  )
}

Experience.defaultProps = {
  experience: undefined,
}

Experience.propTypes = {
  experience: PropTypes.shape({
    entreprise: PropTypes.string.isRequired,
    dateDebut: PropTypes.string.isRequired,
    dateFin: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    intitule: PropTypes.string.isRequired,
    technologies: PropTypes.string.isRequired,
  }),
}

export default Experience
