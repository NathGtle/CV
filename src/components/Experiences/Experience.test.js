import React from 'react'
import renderer from 'react-test-renderer'
import Experience from './Experience'

const mocks = {
  experience: {
    entreprise: 'Groupe Renault',
    intitule: 'Ingénieur Etude et développement',
    dateDebut: 'Janvier 2018',
    dateFin: "Aujourd'hui",
    description: '- Création de module & gestion des sites du groupe Renault)\n',
    technologies:
      'Technologies utilisées: HTML 5, CSS 3, Javascript, Docker\nFramework/Librairies utilisés: React, Vue.JS, Svelte, Node',
  },
}

describe('Experience component', () => {
  it('should render correctly with props', () => {
    const component = renderer.create(<Experience {...mocks} />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly without props', () => {
    const component = renderer.create(<Experience />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
