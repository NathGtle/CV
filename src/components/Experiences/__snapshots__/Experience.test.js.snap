// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience component should render correctly with props 1`] = `
<article
  className="exp"
>
  <div
    className="headline"
  >
    Groupe Renault
  </div>
  <div
    className="duration"
  >
    Janvier 2018
     à 
    Aujourd'hui
  </div>
  <div
    className="intitule"
  >
    Ingénieur Etude et développement
  </div>
  <div
    className="description display-linebreak"
  >
    - Création de module & gestion des sites du groupe Renault)

  </div>
  <div
    className="technologies"
  >
    Technologies utilisées: HTML 5, CSS 3, Javascript, Docker
Framework/Librairies utilisés: React, Vue.JS, Svelte, Node
  </div>
</article>
`;

exports[`Experience component should render correctly without props 1`] = `null`;
